File: algorithms/binary_search/binary_search.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while (low <= high):
        mid = math.floor((high - low) / 2) + low
        if arr[mid] == target:
            return mid
        if arr[mid] < target:
            low = mid + 1
        else: # arr[mid] > target
            high = mid - 1
    return -1

File: algorithms/binary_search/binary_search.js
const binary_search = (arr, target) => {
    let low = 0,
        high = arr.length - 1;
    
    while (low <= high) {
        let mid = Math.floor((high - low) / 2) + low;
        if (arr[mid] == target) {
            return mid
        }
        if (arr[mid] < target) {
            low = mid + 1;
        } else { // arr[mid] > target
            high = mid - 1;
        }
    }
    return -1;
}